control-admin/src/test/java/fi/nls/oskari/control/admin/LayerAdminUsageCheckHandlerTest.java

```diff
@@ -88,13 +100,21 @@ public void testLayerUsageCheckWhenUserIsNotAdmin() {
         Map requestParameters = getRequestParams();
         ActionParameters params = createActionParams(requestParameters, getNotAdminUser());
         try {
-            handler.handleAction(params);
+            adminUsageCheckHandler.handleAction(params);
             fail("ActionDeniedException should have been thrown");
         } catch (ActionException e) {
             assertEquals("Admin only", e.getMessage());
         }
     }
 
+    @Test
+    public void testCleanupLayerReferences() throws Exception {
```

Comment:
I think this should be on it's own test file but it would require similar mocking for both tests. Maybe sharing a helper class so that mocks only need to be written one time?

Reply:
Yeah, a helper class or a shared new common superclass could do the trick here.